infrastructure/merklemountainrange/src/merklemountainrange.rs

```diff
@@ -332,3 +414,31 @@ pub fn get_node_height(index: usize) -> usize {
     let new_index = index - height_index - 1;
     get_node_height(new_index)
 }
+
+/// This function will convert the given index and get its location in the MMR, this only works for leaf nodes
+pub fn get_leaf_index(index: usize) -> usize {
```

Comment:
I think we should disambiguate parameter names that refer to a _node index_ (i.e. any node in the tree) and _object index_ (the index of the data object in the underlying array). Perhaps `node_index` and `object_index` will do fine.

Calling both `index` is going to lead to mistakes somewhere down the line.

Reply:
Done, I tried to also replace all pub fn having a parameter index to reflect this as well